### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by bonifatio**
I propose faster and simpler enum format derivation via Shapeless plus did some refactoring. However I need some help to make it work with _-Xlint_ turned on (currently compiler reports not used `gen: LabelledGeneric.Aux[A, Repr]` if enabled)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -21,7 +21,7 @@ val commonSettings = Seq(
     "-unchecked",
     "-deprecation",
     "-Xfatal-warnings",
-    "-Xlint",
+    //"-Xlint",
```
Comment (t3hnar) [inline build.sbt]: just remove `"-Xfatal-warnings"`

**Comment by t3hnar** [comment]
@fr3akX could you please take a look ?

File: build.sbt (hunk 1 above)
Comment (bonifatio) [reply build.sbt]: All settings are restored

**Comment by t3hnar** [comment]
released in `v0.3.0`

**Pull request merged**